substrate/client/src/light/fetcher.rs

```diff
@@ -78,11 +103,27 @@ impl<S, E, F> LightDataChecker<S, E, F> {
 impl<S, E, F, Block> FetchChecker<Block> for LightDataChecker<S, E, F>
 	where
 		Block: BlockT,
+		Block::Hash: Into<[u8; 32]>,
 		S: BlockchainStorage<Block>,
 		E: CodeExecutor,
 		F: Fetcher<Block>,
 {
-	fn check_execution_proof(&self, request: &RemoteCallRequest<Block::Hash>, remote_proof: Vec<Vec<u8>>) -> ClientResult<CallResult> {
+	fn check_read_proof(
+		&self,
+		request: &RemoteReadRequest<Block::Hash>,
+		remote_proof: Vec<Vec<u8>>
+	) -> ClientResult<Option<Vec<u8>>> {
+		let local_header = self.blockchain.header(BlockId::Hash(request.block))?;
```

Comment:
since this happens well beyond the point where the storage request is actually issued, could the header be pruned if it takes a long time to serve the request or if the header was on the verge of being pruned when the request was issued?